app/scripts/modules/core/src/presentation/hooks/useData.hook.ts

```diff
@@ -15,15 +15,17 @@ import { useLatestPromise, IUseLatestPromiseResult } from './useLatestPromise.ho
  *
  * @param callback the callback to be invoked whenever dependencies change
  * @param defaultResult the default result returned before the first promise resolves
+ *                      this value will be memoized on the first render
  * @param deps array of dependencies, which (when changed) cause the callback to be invoked again
  * @returns an object with the result and current status of the promise
  */
 export function useData<T>(callback: () => PromiseLike<T>, defaultResult: T, deps: any[]): IUseLatestPromiseResult<T> {
+  const memoizedDefaultResult = useMemo(() => defaultResult, []);
```

Comment:
Should we consider providing an option for dynamic `defaultResult`? It may be typed as `T | () => T` and if it is the latter, then we skip the memoization?

Reply:
opinion: I'd rather wait until we find an actual use case where this is necessary before expanding the api